scripts/assets/setup-as-service.ps1

```diff
@@ -0,0 +1,76 @@
+#Requires -RunAsAdministrator
+$ErrorActionPreference = "Stop";
+$ProgressPreference = "SilentlyContinue";
+
+function Get-ScriptDirectory
+{
+    $Invocation = (Get-Variable MyInvocation -Scope 1).Value;
+    Split-Path $Invocation.MyCommand.Path;
+}
+
+function CheckPortIsClosed($port) {
+    $result = Test-NetConnection -Port $port -ComputerName 127.0.0.1 -InformationLevel Quiet 3> $null
+    return $result -eq $false
+}
+
+$scriptDirectory = Get-ScriptDirectory;
+$settingsJson = "settings.json";
+$rvn = "rvn.exe";
+$serverDir = Join-Path $scriptDirectory "Server"
+$settingsJsonPath = Join-Path $serverDir $settingsJson;
+
+$name = 'RavenDB'
+$secure = Read-Host -Prompt 'Would you like to setup a secure server? (y/n)'
+
+if ($secure -match '^\s*?[yY]') {
+    $port = 443
+} else {
+    $port = 8080
+}
+
+if ($port -lt 0 -Or $port -gt 65535){
+    Write-Error "Error. Port must be in the range 0-65535."
+    exit 1
+}
+
+if ((CheckPortIsClosed $port) -eq $false) {
+    Write-Error "Port $port is not available.";
+    exit 2
+}
+
+try
+{
+    $json = Get-Content $settingsJsonPath -raw | ConvertFrom-Json
+    $json.ServerUrl="http://127.0.0.1:$port"
+    $json | ConvertTo-Json  | Set-Content $settingsJsonPath
+}
+catch
+{
+    Write-Error $_.Exception
+    exit 3
+}
+
+Push-Location $serverDir;
+
+Try
+{
+    Invoke-Expression -Command ".\$rvn windows-service register --service-name $name";
```

Comment:
As far as port blocking on Setup Wizard is concerned I tried adding `--ServerUrl=...` to the rvn command end. It fixes the port number on the UI, yet leaves us with unusable service since it adds it to the CLI args of the service (which is by design). After setup it's left with http, while it should have https, so there's a conflict in what we pass to RVN to setup the service and the state after setup.

Reply:
We can ask the user to choose between HTTP / HTTPS at the script level.
But we DON'T want to pass the arguments on the command line. We'll use the `settings.json` file, instead, exactly so we can re-write it easily.

Reply:
OK then, that's what it does. Just wanted to make sure we don't want to go that way.